src/labkey/Exp.ts

```diff
@@ -0,0 +1,915 @@
+ * To perform a file upload over HTTP:
+ *
+ * ```html
+ * <form id="upload-run-form" enctype="multipart/form-data" method="POST">
+ *     <div id="upload-run-button"></div>
+ * </form>
+ * <script type="text/javascript">
+ *     LABKEY.Utils.requiresScript("FileUploadField.js");
+ *     Ext.onReady(function() {
+ *        var form = new Ext.form.BasicForm(
+ *        Ext.get("upload-run-form"), {
+ *           fileUpload: true,
+ *           frame: false,
+ *           // Optional - specify a protocolId so that the Exp.Data object is assigned the related LSID namespace.
+ *           url: LABKEY.ActionURL.buildURL("assay", "assayFileUpload", undefined, { protocolId: 50 }),
+ *           listeners: {
+ *              actioncomplete : function (form, action) {
+ *                 alert('Upload successful!');
+ *                 var data = new LABKEY.Exp.Data(action.result);
+ *
+ *                 // now add the data as a dataInput to a LABKEY.Exp.Run
+ *                 var run = new LABKEY.Exp.Run();
+ *                 run.name = data.name;
+ *                 run.dataInputs = [ data ];
+ *
+ *                 // add the new run to a LABKEY.Exp.Batch object and
+ *                 // fetch the parsed file contents from the data object
+ *                 // using the LABKEY.Exp.Data#getContent() method.
+ *              },
+ *              actionfailed: function (form, action) {
+ *                 alert('Upload failed!');
+ *              }
+ *           }
+ *        });
+ *
+ *        var uploadField = new Ext.form.FileUploadField({
+ *           id: "upload-run-field",
+ *           renderTo: "upload-run-button",
+ *           buttonText: "Upload Data...",
+ *           buttonOnly: true,
+ *           buttonCfg: { cls: "labkey-button" },
+ *           listeners: {
+ *              "fileselected": function (fb, v) {
+ *                 form.submit();
+ *              }
+ *           }
+ *        });
+ *     });
+ * </script>
+ * ```
+ *
+ * Or, to upload the contents of a JavaScript string as a file:
+ *
+ * ```js
+ * Ext.onReady(function() {
+ *  LABKEY.Ajax.request({
+ *    url: LABKEY.ActionURL.buildURL("assay", "assayFileUpload"),
+ *    params: { fileName: 'test.txt', fileContent: 'Some text!' },
+ *    success: function(response, options) {
+ *       var data = new LABKEY.Exp.Data(Ext.util.JSON.decode(response.responseText));
+ *
+ *       // now add the data as a dataInput to a LABKEY.Exp.Run
+ *       var run = new LABKEY.Exp.Run();
+ *       run.name = data.name;
+ *       run.dataInputs = [ data ];
+ *
+ *       // add the new run to a LABKEY.Exp.Batch object here
+ *    }
+ *  });
+ * });
+ * ```
+ */
+export class Data extends ExpObject {
+
+    /**
+     * The DataClass the data belongs to.
+     */
+    dataClass: ExpDataDataClass;
+
+    /**
+     * The local file url of the uploaded file.
+     */
+    dataFileURL: string;
+
+    /**
+     * TODO: Describe dataType
+     */
+    dataType: string;
```

Comment:
@labkey-kevink Is `dataType` a thing on `Exp.Data`? Where does it reside? I couldn't find it in platform. Most properties are found in `ExperimentJSONCoverter.java`.

Reply:
You're right, it looks like we never included it -- we can remove it from Exp.ts.  There is an ExpData.getDataType() but we aren't serializing it ExperimentJSONConverter.